BiometricLocker/BiometricLocker.swift

```diff
@@ -70,49 +97,92 @@ open class BiometricLocker {
+    private var notificationObservers = [NSObjectProtocol]()
+
+    public init(localizedReason: String, automaticallyLocksOnBackgroundOrQuit: Bool = true, withUnlockedTimeAllowance timeAllowance: TimeInterval = LATouchIDAuthenticationMaximumAllowableReuseDuration) {
+        self.localizedReason = localizedReason
+        
+        self.unlockedTimeAllowance = timeAllowance
+
+        if automaticallyLocksOnBackgroundOrQuit {
+            self.notificationObservers.append(NotificationCenter.default.addObserver(forName: .UIApplicationDidEnterBackground, object: nil, queue: .main) { _ in
+                // If app goes into background, we start the clocks to lock the app.
+                self.lock()
+            })
+
+            self.notificationObservers.append(NotificationCenter.default.addObserver(forName: .UIApplicationWillTerminate, object: nil, queue: .main) { _ in
+                // If the app is killed, lock it instantly.
+                self.lock(.now)
+            })
+        }
+    }
+
+    deinit {
+        self.notificationObservers.forEach { observer in
+            NotificationCenter.default.removeObserver(observer)
+        }
+    }
+
+    /**
+     Tells the biometric locker to lock the app. Defaults to the currently defined time allowance (5min by default).
+
+     - Important: To change when the app locks after having called this, simply call `lock` again with a different allowance, or tell the app to `lock(.afterTimeInterval(timeInterval)`, as it supercedes the current unlocked time allowance. `unlockedTimeAllowance` is only checked when we ask the BiometricLocker whether the app is locked or not. Changing it before or after calling `lock` should work just fine.
+     - Parameter when: An enum defining when the app should lock. Now, after the time allowance, or after a custom time interval.
+     */
+    public func lock(_ when: LockingTime = .afterTimeAllowance) {
         // if we are already deactivated, return. Otherwise you can just kill the app and try again to bypass touch ID.
         if self.isLocked { return }
+
+        let date: Date
+        switch when {
+        case .now:
+            date = Date.distantPast
+        case .afterTimeAllowance:
+            date = Date()
+        case .afterTimeInterval(let interval):
+            date = Date(timeIntervalSinceNow: (interval - self.unlockedTimeAllowance))
+        }
+
         self.defaults.set(date, forKey: Key.applicationDidEnterBackgroundDate.rawValue)
         self.defaults.synchronize()
     }
 
     /// Unlocks the app.
     ///
-    /// Call it from the `AuthenticationDelegate`'s `didAuthenticateSuccessfully` method, or if the user session was destroyed. (No sense in locking the app if the user logs out).
-    open func unlock() {
+    /// - Important: Should be called if the user session is detroyed, if your app has them, otherwise you risk having the app locked whislt there's no user, preventing people from logging in / signing up.
```

Comment:
still a couple spelling issues here: 
- `detroyed` -> `destroyed` 
- `whislt` -> `whilst` or `while`

Reply:
Hard to believe I type stuff for a living. 😂